hydra/utils.py

```diff
@@ -18,32 +18,24 @@ def get_method(path: str) -> type:
 
 def get_class(path: str) -> type:
     try:
-        from importlib import import_module
-
-        module_path, _, class_name = path.rpartition(".")
-        mod = import_module(module_path)
-        try:
-            klass: type = getattr(mod, class_name)
-        except AttributeError:
-            raise ImportError(
-                "Class {} is not in module {}".format(class_name, module_path)
-            )
+        from pydoc import locate
+
+        klass = locate(path)
```

Comment:
This can't be implemented with importlib?

Reply:
Under the hood it is implemented with `__import__` but with a bunch of logic to get the class methods and to recursive import the correct module.  I believe `importlib` is just a friendly wrapper around `__import__`.  You can see the code for [locate](https://github.com/python/cpython/blob/master/Lib/pydoc.py#L1620) and [safeimport (the underlying importer)](https://github.com/python/cpython/blob/master/Lib/pydoc.py#L318).  One could implement this with `importlib`. but it's not going to be as good as this implementation.

Reply:
I ended up re-implementing and simplifying the pydoc function because there was some strange behavior if a submodule on the chain doesn't load.